lib/elixir/src/elixir_expand.erl

```diff
@@ -1287,6 +1297,10 @@ format_error({no_parens_nullary_remote, Remote, Fun}) ->
   io_lib:format("missing parenthesis on call to ~ts.~ts/0. "
                 "parenthesis are always required on function calls without arguments",
                 [elixir_aliases:inspect(Remote), Fun]);
+format_error({parens_map_lookup_guard, Map, Field}) ->
+  io_lib:format("map lookup do not allow parenthesis inside guards "
+                "and cannot invoke remote function where the module is a variable: ~ts.~ts()",
```

Comment:
What do you think about:

    cannot invoke remote function in guard. If you wanted to do a map lookup instead, please remove parens